### PR TITLE
fixing deprecated warning, ticket #12472

### PR DESCRIPTION
**Pull request by aleksandra-tarkowska**
this should resolve deprecation warning. To test just see if searching works

### CODE REVIEW

**Comment by joshmoore** [comment]
Searching for `a*` at https://trout.openmicroscopy.org/merge/webclient/search/
causes:
```
  File "/opt/hudson/workspace/OMERO-5.1-merge-deploy/src/dist/lib/python/omeroweb/webclient/controller/search.py", line 100, in search
    self.searchError = "Your query for '%s' caused an error: %s." % (query, x.reason)

AttributeError: 'InternalException' object has no attribute 'reason'
```

@aleksandra-tarkowska : did you try using https://docs.python.org/2/library/warnings.html ?

**Comment by aleksandra-tarkowska** [comment]
My initial testing was based on raising exception within the code. It looks like due to change in https://github.com/openmicroscopy/openmicroscopy/blob/develop/components/tools/OmeroPy/src/omero/gateway/__init__.py#L4102 we don't see deprecated warning anymore. I reverted to the original changes, but removed other as we only case about maxClauseCount

**Comment by aleksandra-tarkowska** [comment]
the same is here:
 - https://github.com/openmicroscopy/openmicroscopy/blob/develop/components/tools/OmeroWeb/omeroweb/webadmin/views.py#L577
 - https://github.com/openmicroscopy/openmicroscopy/blob/develop/components/tools/OmeroWeb/omeroweb/webadmin/views.py#L584
 - https://github.com/openmicroscopy/openmicroscopy/blob/develop/components/tools/OmeroWeb/omeroweb/webclient/views.py#L2760
 - https://github.com/openmicroscopy/openmicroscopy/blob/develop/components/tools/OmeroWeb/omeroweb/webclient/views.py#L3118
 - https://github.com/openmicroscopy/openmicroscopy/blob/develop/components/tools/OmeroWeb/omeroweb/webclient/views.py#L3269
 - https://github.com/openmicroscopy/openmicroscopy/blob/develop/components/tools/OmeroWeb/omeroweb/webgateway/marshal.py#L131

File: components/tools/OmeroWeb/omeroweb/webclient/controller/search.py [hunk 1]
```diff
@@ -97,10 +97,6 @@ def doSearch(searchType):
                     self.containers[dt] = doSearch(dt)
                     resultCount += len(self.containers[dt])
         except Exception, x:
-            msg = x.message
-            if "TooManyClauses" in msg:
+            if hasattr(x, "message") and "TooManyClauses" in x.message:
                 self.searchError = "Please try to narrow down your query. The wildcard matched too many terms."
-            else:
```
Comment (joshmoore) [inline components/tools/OmeroWeb/omeroweb/webclient/controller/search.py]: Why is this removed?

**Comment by aleksandra-tarkowska** [comment]
Sorry just a mistake

**Comment by aleksandra-tarkowska** [comment]
All omero.ServerError exceptions are safe as https://github.com/openmicroscopy/openmicroscopy/blob/develop/components/blitz/resources/omero/ServerErrors.ice#L123
We may want to add some helper wrapper to validate which type of exceptions we are getting then we will have a better way of controlling it in one place

**Comment by aleksandra-tarkowska** [comment]
All ```omero.ServerError``` exceptions are safe as https://github.com/openmicroscopy/openmicroscopy/blob/develop/components/blitz/resources/omero/ServerErrors.ice#L123
We may want to add some helper wrapper to validate which type of exceptions we are getting, then we will have a better way of controlling it in one place. Something like:

```
class WebError(Exception):

    def __init__(self, exc):
        Exception.__init__(self, exc)
        if isinstance(exc, omero.ServerError):
            self._message = exc.message
        else:
            self._message = exc
        
    message = property(lambda self: self._message, 
                       lambda self, v: setattr(self, '_message', v))

    def __unicode__(self):
        return unicode(self.message)
```

@will-moore @knabar @chris-allan @sbesson @manics any thoughts?

**Comment by manics** [comment]
Could you use `str(x)` for everything?

**Comment by aleksandra-tarkowska** [comment]
Yes I did, but then you end up with the whole stack-trace in UI message. That is a maker problem

**Comment by joshmoore** [comment]
Thanks for battling through this one, @aleksandra-tarkowska. New strategy makes sense to me (without having tested it yet)

**Comment by joshmoore** [comment]
```
Please try to narrow down your query. The wildcard matched too many terms.
No results found
```
Assuming there's no easy way to get rid of the "No results found" on the page, I'd say :thumbsup: @aleksandra-tarkowska, what do you think?

**Comment by aleksandra-tarkowska** [comment]
That is default icon. We would need to design new one for error. At this moment we really capture 1 error. Is that worth?

**Comment by joshmoore** [comment]
Nope. If there's not something easy (like "returning empty string" or whatever), then let's not worry about it. Thanks.
